File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.length();
        set<char> st;
        int i=0,j=0,ans=0;
        while(i<n and j<n){
            if(!st.count(s[j])){
                st.insert(s[j]);
                ans=max(ans,(int)st.size());
                j++;
            }
            else{
                st.erase(s[i]);
                i++;
            }
        }
        return ans;
    }
};

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i,j,ans=0,0,0
        n=len(s)
        seen=set();
        while i<n and j<n:
            if s[j] not in seen:
                seen.add(s[j])
                ans=max(ans,int(len(seen)))
                j+=1
            else:
                seen.remove(s[i])
                i+=1
        return ans